src/main/java/pl/edu/pw/mini/ingreedio/api/repository/impl/CustomizedProductRepositoryImpl.java

```diff
@@ -1,48 +1,47 @@
 package pl.edu.pw.mini.ingreedio.api.repository.impl;
 
 import java.util.List;
-import java.util.Objects;
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.mongodb.core.MongoTemplate;
 import org.springframework.data.mongodb.core.query.Criteria;
 import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.stereotype.Repository;
 import pl.edu.pw.mini.ingreedio.api.criteria.ProductFilterCriteria;
-import pl.edu.pw.mini.ingreedio.api.dto.FullProductDto;
-import pl.edu.pw.mini.ingreedio.api.mapper.FullProductDtoMapper;
+import pl.edu.pw.mini.ingreedio.api.dto.ProductDto;
+import pl.edu.pw.mini.ingreedio.api.mapper.ProductDtoMapper;
 import pl.edu.pw.mini.ingreedio.api.model.Product;
 import pl.edu.pw.mini.ingreedio.api.repository.CustomizedProductRepository;
 
 @RequiredArgsConstructor
 @Repository
 public class CustomizedProductRepositoryImpl implements CustomizedProductRepository {
     private final MongoTemplate mongoTemplate;
-    private final FullProductDtoMapper fullProductDtoMapper;
+    private final ProductDtoMapper productDtoMapper;
 
     @Override
-    public List<FullProductDto> filterProducts(ProductFilterCriteria criteria) {
+    public List<ProductDto> getProductsMatching(ProductFilterCriteria criteria) {
         Query query = new Query();
 
-        if (!Objects.isNull(criteria.getName())) {
+        if (criteria.getName() != null) {
```

Comment:
you told me yourself to use Objects.isNull... two-faced much....

Reply:
truewww but i did done some research and Objects.isNull should only be used as predicate, I'm so sorry diva for that...